ciao-controller/main.go

```diff
@@ -283,3 +295,33 @@ func (c *controller) startCiaoService() error {
 
 	return nil
 }
+
+func getControllerFQDNFromCert(cert string) (string, error) {
+	bytesCert, err := ioutil.ReadFile(cert)
+	if err != nil {
+		return "", fmt.Errorf("couldn't read cert: %s", err)
+	}
+
+	blockCert, _ := pem.Decode(bytesCert)
+	if blockCert == nil {
+		return "", fmt.Errorf("couldn't decode cert")
+	}
+
+	parsedCert, err := x509.ParseCertificate(blockCert.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("couldn't parse cert: %s", err)
+	}
+
+	role := ssntp.GetRoleFromOIDs(parsedCert.UnknownExtKeyUsage)
+	if !role.IsController() {
+		return "", fmt.Errorf("expected certificate role ssntp.Controller, got role %s", role.String())
```

Comment:
Or not.  For that side of the house we have a CACert and a _key_.

The CA cert is only optionally created by us and in production I expect it would come from the cluster operator's IT certificate authority, not us.

The key is not a x509 struct.  I need to look into whether we could embed metadata in that...in general I suspect it's possible given my gpg key has a hunk of metadata that even includes a jpeg of me!

But even if we did a pivot to embed the desired information in this key, we'd still need to tweek the deploy automation to do this for that key in addition to having ciao-cert do it for the ssntp cert's.  AND we'd need to know that the embed works as an after the fact add-on, in case an operator's IT org hands them the key to use which was signed by the IT org's CAcert, not the operator creates the cert and gets it signed by the org's CAcert.  I think is something we must expect for a real production deployment.

Soooo...I think the ssntp cert is an easier route to go for now.  We're really just looking for a name and there is a name in the ssntp cert.  As long as the ssntp cert was created with reference to the external name of the controller, it can be a reliable source of the needed name.  Then we don't have to try to interact with the operator to coordinate modifications to a key that is signed by their org.